### PR TITLE
Stage Voice Channel Create

### PR DESCRIPTION
**Pull request by dbmIsaac**
Created Stage voice channel action

**Please describe the changes this PR makes and why it should be merged:**
Created a action to create stage voice channels. 
**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [ ] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by DetroitWhiskey136 — CHANGES_REQUESTED** [review]
You'll have to sort out your linting errors before proceeding :)

File: actions/create_stage_voice_channel.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  // This is the section the action will fall into.
+  // ---------------------------------------------------------------------
+
+  section: 'Channel Control',
+
+  // ---------------------------------------------------------------------
+  // Action Subtitle
+  //
+  // This function generates the subtitle displayed next to the name.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:22]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  subtitle(data, presets) {
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:24]:
```suggestion
  subtitle(data) {
```

File: actions/create_stage_voice_channel.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    return `${data.channelName}`;
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Storage Function
+  //
+  // Stores the relevant variable info for the editor.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:32]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Meta Data
+  //
+  // Helps check for updates and provides info if a custom mod.
+  // If this is a third-party mod, please set "author" and "authorUrl".
+  //
+  // It's highly recommended "preciseCheck" is set to false for third-party mods.
+  // This will make it so the patch version (0.0.X) is not checked.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:48]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+  meta: { version: '2.1.7', preciseCheck: true, author: null, authorUrl: null, downloadUrl: null },
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:50]:
```suggestion
  meta: { version: '2.1.7', preciseCheck: false, author: 'DBM Mods', authorUrl: 'https://github.com/dbm-network/mods', downloadUrl: 'https://github.com/dbm-network/mods/blob/master/actions/create_stage_voice_channel.js' },
```

File: actions/create_stage_voice_channel.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+  // ---------------------------------------------------------------------
+  // Action Fields
+  //
+  // These are the fields for the action. These fields are customized
+  // by creating elements with corresponding IDs in the HTML. These
+  // are also the names of the fields stored in the action's JSON data.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:58]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  // Command HTML
+  //
+  // This function returns a string containing the HTML used for
+  // editing actions.
+  //
+  // The "isEvent" parameter will be true if this action is being used
+  // for an event. Due to their nature, events lack certain information,
+  // so edit the HTML to reflect this.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:71]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+  html(isEvent, data) {
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:73]:
```suggestion
  html() {
```

File: actions/create_stage_voice_channel.js [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+<store-in-variable allowNone selectId="storage" variableInputId="varName" variableContainerId="varNameContainer"></store-in-variable>`;
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Editor Init Code
+  //
+  // When the HTML is first applied to the action editor, this code
+  // is also run. This helps add modifications or setup reactionary
+  // functions for the DOM elements.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:110]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+  init() {},
+
+  // ---------------------------------------------------------------------
+  // Action Bot Function
+  //
+  // This is the function for the action within the Bot's Action class.
+  // Keep in mind event calls won't have access to the "msg" parameter,
+  // so be sure to provide checks for variable existence.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:120]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+  action(cache) {
+    const data = cache.actions[cache.index];
+    const server = cache.server;
+    if (!server?.channels) return this.callNextAction(cache);
+    const name = this.evalMessage(data.channelName, cache);
+    const storage = parseInt(data.storage, 10);
+    const reason = this.evalMessage(data.reason, cache);
+    /** @type {import('discord.js').GuildChannelCreateOptions} */
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:129]:
```suggestion

```

File: actions/create_stage_voice_channel.js [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Bot Mod
+  //
+  // Upon initialization of the bot, this code is run. Using the bot's
+  // DBM namespace, one can add/modify existing functions if necessary.
+  // In order to reduce conflicts between mods, be sure to alias
+  // functions you wish to overwrite.
+  // ---------------------------------------------------------------------
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:160]:
```suggestion
```

**Review by TheMonDon — CHANGES_REQUESTED** [review]
Commit all of these changes and then I will see if there are any other changes needed.

File: actions/create_stage_voice_channel.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+  meta: {
+    version: '2.1.7',
+    preciseCheck: false,
+    author: 'DBM Mods',
+    authorUrl: 'https://github.com/dbm-network/mods',
+    downloadUrl: 'https://github.com/dbm-network/mods/blob/master/actions/create_stage_voice_channel.js',
+  },
+
+  fields: ['channelName', 'bitrate', 'userLimit', 'varName', 'categoryID', 'reason'],
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:24]:
```suggestion
  fields: ['channelName', 'categoryID', 'bitrate', 'userLimit', 'reason', 'storage', 'varName'],
```

File: actions/create_stage_voice_channel.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
+module.exports = {
+  name: 'Create Stage Voice Channel',
+
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:3]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+      channelData.userLimit = parseInt(this.evalMessage(data.userLimit, cache), 10);
+    }
+    if (data.categoryID) {
+      channelData.parent = this.evalMessage(data.categoryID, cache);
+    }
+    server.channels
+      .create(name, {
+        ...channelData,
+        type: 'GUILD_STAGE_VOICE',
+      })
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:80]:
```suggestion
      .create(name, channelData)
```

File: actions/create_stage_voice_channel.js [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+  action(cache) {
+    const data = cache.actions[cache.index];
+    const server = cache.server;
+    if (!server?.channels) return this.callNextAction(cache);
+    const name = this.evalMessage(data.channelName, cache);
+    const storage = parseInt(data.storage, 10);
+    const reason = this.evalMessage(data.reason, cache);
+    const channelData = { reason };
+    if (data.bitrate) {
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:67]:
```suggestion

    if (data.reason) {
      channelData.reason = this.evalMessage(data.reason, cache);
    }
    if (data.bitrate) {
```

File: actions/create_stage_voice_channel.js [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+
+  init() {},
+
+  action(cache) {
+    const data = cache.actions[cache.index];
+    const server = cache.server;
+    if (!server?.channels) return this.callNextAction(cache);
+    const name = this.evalMessage(data.channelName, cache);
+    const storage = parseInt(data.storage, 10);
+    const reason = this.evalMessage(data.reason, cache);
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:65]:
```suggestion
```

File: actions/create_stage_voice_channel.js [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    const channelData = { reason };
```
Comment (TheMonDon) [inline actions/create_stage_voice_channel.js:66]:
```suggestion
    const channelData = { type: 'GUILD_STAGE_VOICE' };
```

**Review by TheMonDon — CHANGES_REQUESTED** [review]

**Comment by TheMonDon** [comment]
Closed on request of author

**Pull request closed without merging**